src/util/hash.rs

```diff
@@ -21,8 +21,39 @@ use util::pair::*;
 
 use crypto::ripemd160::Ripemd160;
 use crypto::sha2::Sha256;
+use crypto::sha3::Sha3;
 use crypto::digest::Digest;
 
+#[derive(Serialize, Deserialize)]
+pub struct Keccak256Hash(pub [u8; 32]);
+impl Keccak256Hash {
+    pub fn from_data(data: &[u8]) -> Keccak256Hash {
+        let mut ret = [0u8; 32];
+        let mut sha3 = Sha3::keccak256();
+        sha3.input(data);
+        sha3.result(&mut ret);
+        Keccak256Hash(ret)
+    }
+}
+impl_array_newtype!(Keccak256Hash, u8, 32);
+impl_array_hexstring_fmt!(Keccak256Hash);
+impl_byte_array_newtype!(Keccak256Hash, u8, 32);
+
+#[derive(Serialize, Deserialize)]
+pub struct Sha256Hash(pub [u8; 32]);
```

Comment:
This shouldn't be part of this diff -- it's already defined in my branch.

Reply:
Was hoping your branch would be merged in dev before this PR, I'm happy to wait for that or branch this off of your branch.

Reply:
Now using your def in the dev branch